cmd/determinize-ci-operator/main.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"flag"
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/ci-operator-prowgen/pkg/config"
+	"github.com/openshift/ci-operator-prowgen/pkg/promotion"
+	"github.com/openshift/ci-operator/pkg/api"
+)
+
+func gatherOptions() promotion.Options {
+	o := promotion.Options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	o.Bind(fs)
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatal("could not parse input")
+	}
+	return o
+}
+
+func main() {
+	o := gatherOptions()
+	if err := o.Validate(); err != nil {
+		logrus.Fatalf("Invalid options: %v", err)
+	}
+
+	var toCommit []config.Info
+	if err := config.OperateOnCIOperatorConfigDir(o.ConfigDir, func(configuration *api.ReleaseBuildConfiguration, repoInfo *config.FilePathElements) error {
+		if (o.Org != "" && o.Org != repoInfo.Org) || (o.Repo != "" && o.Repo != repoInfo.Repo) {
+			return nil
+		}
+		if !(promotion.PromotesOfficialImages(configuration) && configuration.PromotionConfiguration.Name == o.CurrentRelease) {
```

Comment:
I wonder whether a generically-sounding `determinize-*` tool should have this hard-coded. I understand this was actually needed only for those configs in this case, but maybe at least put it behind a flag?

Reply:
We only ever need to determinize configs we'd write to or generate, which this is the definition of. I don't see a reason to run the tool on other configs right now. In the future maybe we can make it more generic if we need it to be?

Reply:
I would say we don't need generic determinize right now - this is about getting ci-operator and prow configs right only.